about/about-kubernetes-ingress.md

```diff
@@ -0,0 +1,130 @@
+
+To get your ingress traffic to the in-cluster ingress pods, the ingress pods are normally exposed externally as a
+Kubernetes service, and any of the standard ways of accessing the service from outside of the cluster can be used. A
+common approach is use an external network load balancer, or service IP advertisement, with
+`externalTrafficPolicy:local` to minimize the number of network hops, and to retain client source IP address, which
+allows network policy to be used to restrict access to the ingress pods to particular clients if desired.
+
+![In-cluster ingress]({{site.baseurl}}/images/ingress-in-cluster.svg)
+
+#### External ingress solutions
+
+External ingress solutions use application load balancers outside of the cluster. The exact details and
+features depend on which ingress controller you are using, but most cloud providers include an ingress controller that
+automates the provisioning and management of the cloud provider's application load balancers to provide ingress.
+
+The advantages of this type of ingress solutions are that your cloud provider handles the operational complexity of the
+ingress for you.  The downsides are a potentially more limited set of features compared to the rich range of in-cluster
+ingress solutions, and the maximum number of services exposed by ingress being constrained by cloud provider specific
+limits.
+
+![External ingress]({{site.baseurl}}/images/ingres-external.svg)
+
+Note that most application load balancers support a basic mode of operation of forwarding traffic to the chosen service
+backing pods via the [node port]({{site.baseurl}}/about/about-kubernetes-services#node-port-services) of the
+corresponding service.
+
+In addition to this basic approach of load balancing to service node ports, some cloud providers support a second mode
+of application layer load balancing, which load balances directly to the pods backing each service, without going via
+node-ports or other kube-proxy service handling. This has the advantage of eliminating the potential second network hop
+associated with node ports load balancing to a pod on a different node. The potential disadvantage is that if you are
+operating at very high scales, for example with hundreds of pods backing a service, you may exceed the application layer
+load balancers maximum limit of IPs it can load balance to in this mode. In this case switching to an in-cluster ingress
+solution is likely the better fit for you.
+
+### Show me everything!
+
+All the above diagram focus on connection level (L5-7) representation of ingress and services. You can learn more about
+the network level (L3-4) interactions involved in handling the connections, including which scenarios client source IP
+addresses are maintained, in the [About Kubernetes Services]({{site.baseurl/about/about-kubernetes-services}}) guide.
+
+If you are already up to speed on how services work under the covers, here's some more complete diagrams show how the details
```

Comment:
If you are already up to speed on how services work under the covers, here are more complete diagrams that show details of how services are load balanced at the network layer (L3-4).